packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -3666,14 +3745,27 @@ describe('NetworkController', () => {
           const fakeNetworkClient = buildFakeClient(fakeProvider);
           createNetworkClientMock.mockReturnValue(fakeNetworkClient);
 
-          try {
-            // @ts-expect-error Intentionally passing invalid type
-            await controller.setProviderType(NetworkType.rpc);
-          } catch {
-            // catch the rejection (it is tested above)
-          }
+          const promiseForNoStateChanges = await waitForStateChanges({
+            messenger,
+            propertyPath: [
+              'networksMetadata',
+              NetworkType.rpc,
+              'details',
+              'EIPS',
+              '1559',
+            ],
+            count: 0,
+            operation: async () => {
+              try {
+                // @ts-expect-error Intentionally passing invalid type
+                await controller.setProviderType(NetworkType.rpc);
+              } catch {
+                // catch the rejection (it is tested above)
+              }
+            },
+          });
 
-          expect(controller.state.networkDetails.EIPS[1559]).toBeUndefined();
+          expect(Boolean(promiseForNoStateChanges)).toBe(true);
```

Comment:
The reason for this is that the `controller.state.networksMetadata[NetworkType.rpc].details` won't exist yet since it gets populated when first actually initialized so we can't check for the state change by statically referencing the state, instead we check that there is no state change op.

Reply:
We could get the networkDetails at the start of the test and compare it to the end of the test and make sure that its the same (not same reference, gotta make a copy and deep compare them). In effect, it would be checking that the only items in networkMeta is the default (mainnet). Might be a stronger way to test the same thing.